src/redux/books/booksSlice.js

```diff
@@ -1,47 +1,76 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+export const fetchBooks = createAsyncThunk('book/fetchBooks', async () => {
+  const response = await axios.get('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/tfn9nyLdsA3a9Lmk93em/books');
+  const bookList = Object.entries(response.data).map(([key, value]) => {
+    const { title, category, author } = value[0];
+    return {
+      Id: key,
+      title,
+      author,
+      category,
+    };
+  });
+  return bookList;
+});
+
+export const newBook = createAsyncThunk('book/addBook', async (book) => {
+  try {
+    await axios.post('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/tfn9nyLdsA3a9Lmk93em/books', book);
+    fetchBooks();
+    return book;
+  } catch {
+    return 'error adding new book';
+  }
+});
+
+export const deleteBook = createAsyncThunk('book/removeBook', async (id) => {
+  await axios.delete(`https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/tfn9nyLdsA3a9Lmk93em/books/${id}`);
+  return id;
+});
 
```

Comment:
- [ ] Even if the create and remove books in a remote server works as expected, however, I'm getting this error message every time I make an action to the remote server or reload the page. Kindly fix this. 👍🏻 

![image](https://user-images.githubusercontent.com/79058364/231661847-35f33b07-dee0-4c9b-8ef3-081b7d1b3c83.png)

Reply:
@BenMukebo I will try to resolve the issue. though I can't see it from my side 🤔  .

Reply:
@lengushuru, You can reclone your app and check if it's still running without any brower error. If so the next CR will check it out and approche this PR